tests/test_prompt.py

```diff
@@ -441,3 +441,33 @@ def test_multiline_defaults(tmp_path_factory, spawn):
         == answers["json_multi"]
         == ["one", "two", {"three": ["four"]}]
     )
+
+
+def test_partial_interrupt(tmp_path_factory, spawn):
+    template, subproject = (
+        tmp_path_factory.mktemp("template"),
+        tmp_path_factory.mktemp("subproject"),
+    )
+    build_file_tree(
+        {
+            template
+            / "copier.yml": yaml.dump(
+                {
+                    "question_1": "answer 1",
+                    "question_2": "answer 2",
+                }
+            ),
+        }
+    )
+    tui = spawn(COPIER_PATH + (str(template), str(subproject)), timeout=10)
+    deque(map(tui.expect_exact, ["question_1?", "Format: str", "answer 1"]))
+    # Answer the first question using the default.
+    tui.sendline()
+    # Send a ctrl-c.
+    tui.send(Keyboard.ControlC + Keyboard.Enter)
+    # This string (or similar) should show in the output if we've gracefully caught the
```

Comment:
@Yajo here's the verification that our stacktrace is being handled gracefully.

Reply:
Incidentally - I enjoyed this test rig after wrapping my head around it. Found it quirky though to get established. It looks like `tui.expect_exact()` consumes stdout until it reads those characters _somewhere_.  Had to dig around the API in pdb to find `readlines()` on the POpenSpawn object which made grokking what was going on under the hood easier.

I'd like to incorporate this in some of my own stuff. Pretty good ROI for contributing to this project :)

Reply:
I'm happy to see you're having a pleasant contribution experience 😊